src/iter.rs

```diff
@@ -631,6 +631,59 @@ pub trait NonEmptyIterator {
             b: other.into_nonempty_iter(),
         }
     }
+
+    /// A specialization of [`NonEmptyIterator::collect`] for convenience.
+    ///
+    /// ```
+    /// use nonempty_collections::*;
+    ///
+    /// let a = nev![1, 2, 3];
+    /// let r = a.into_nonempty_iter().collect_nev();
+    ///
+    /// assert_eq!(nev![1, 2, 3], r);
+    /// ```
+    fn collect_nev(self) -> NEVec<Self::Item>
+    where
+        Self: Sized,
+    {
+        self.collect()
+    }
```

Comment:
I really like the idea of `try_into_nonempty_iter`, but I will ask for these convenience `collect_*` functions to be excluded from the PR. I know it's annoying to have to manually specifying types sometimes, but that matches the existing culture in `std`.

Reply:
Alright, will get rid of them.